continuous-deployment/secrets-management/readme.md

```diff
@@ -1,129 +1,122 @@
-Each developer may maintain their own version of the file or, if required, it may be circulated via private channels e.g. a Teams chat.
+The general approach is to keep secrets in separate configuration files that are not checked in
+to the repo. Add the files to the [.gitignore](https://git-scm.com/docs/gitignore) to prevent that they're checked in.
 
-In a production system, assuming Azure, the secrets would be created in the environment of the running process. This can be done by manually editing the 'Applications Settings' section of the resource but is suggested that a script using
-the Azure CLI to do the same is a useful time-saving utility. See [here](https://docs.microsoft.com/en-us/cli/azure/webapp/config/appsettings?view=azure-cli-latest) for more details.
+Each developer maintains their own local version of the file or, if required, circulate them via private channels e.g. a Teams chat.
 
-*It is best practice to maintain separate secrets configurations for each separate environment that you run. e.g. dev, test, prod, local etc*
-When using a branch-based deployment strategy (e.g. 'master' deploys to production, 'development' deploys to staging, and so forth), a simple way to manage separate secrets configurations on a per-environment basis is described in the [secrets-per-branch recipe](./recipes/azure-devops/secrets-per-branch.md).
+In a production system, assuming Azure, create the secrets in the environment of the running process. We can do this by manually editing the 'Applications Settings' section of the resource, but a script using
+the Azure CLI to do the same is a useful time-saving utility. See [az webapp config appsettings](https://docs.microsoft.com/en-us/cli/azure/webapp/config/appsettings?view=azure-cli-latest) for more details.
 
-### Keeping Secrets Secret
+It's best practice to maintain separate secrets configurations for each environment that you run. e.g. dev, test, prod, local etc
 
-It should be obvious, but all the care taken to protect secrets is all for nothing if we aren't careful about how we use secrets once we have them. Primarily: **DON'T LOG SECRETS!!**. Don't put them in logging or reporting. Don't send them to other applications, don't use them in URLs, forms or in any other way other than to make a request to the service that requires that secret.
+The [secrets-per-branch recipe](./recipes/azure-devops/secrets-per-branch.md) describes a simple way to manage separate secrets configurations for each environment.
 
-### Enhanced-Security Applications
+> Note: even if the secret was only pushed to a feature branch and never merged, it's still a part of the git history. Follow [these instructions](https://help.github.com/en/github/authenticating-to-github/removing-sensitive-data-from-a-repository) to remove any sensitive data and/or regenerate any keys and other sensitive information added to the repo.
```

Comment:
I might also mention that you can rotate the key/secret so that the checked in secret is no longer active.

Reply:
fixed